### PR TITLE
Switched most mixins to events

### PR DESCRIPTION
**Pull request by PotatoPresident**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,5 +1,5 @@
 plugins {
-	id 'fabric-loom' version '0.5-SNAPSHOT'
+	id 'fabric-loom' version '0.6-SNAPSHOT'
```
Comment (yitzy299) [inline build.gradle:2]: Have you tested to make sure loom 0.6 works here? I've seen other people have had issue with it

File: src/main/java/com/github/fabricservertools/deltalogger/events/BlockPlaceCallback.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+					if (result != ActionResult.PASS) {
+						return result;
+					}
+				}
+
+				return ActionResult.PASS;
+			});
+
+	ActionResult place(PlayerEntity player, ItemPlacementContext context);
+}
```
Comment (yitzy299) [inline src/main/java/com/github/fabricservertools/deltalogger/events/BlockPlaceCallback.java:28]: We could likely switch this over to FAPI once fabricmc/fabric#1264 is merged, which adds a block place event

File: src/main/java/com/github/fabricservertools/deltalogger/mixins/BlockInteractionMixin.java [hunk 3]
```diff
@@ -1,96 +0,0 @@
...
-
-    Chat.send(player, placementMessage);
-
-    ret.setReturnValue(ActionResult.SUCCESS);
-    ((ServerPlayerEntity) player).networkHandler.sendPacket(new ScreenHandlerSlotUpdateS2CPacket(
-            -2,
-            player.inventory.selectedSlot,
-            player.inventory.getStack(player.inventory.selectedSlot)));
-  }
-}
```
Comment (yitzy299) [inline src/main/java/com/github/fabricservertools/deltalogger/mixins/BlockInteractionMixin.java:96]: 🦀

**Review by yitzy299 — APPROVED** [review]
Looks fine, thanks for the PR

File: build.gradle (hunk 1 above)
Comment (PotatoPresident) [reply build.gradle:2]: Not extensively. I hadn't heard that. Didn't experience any issues though.

Comment (yitzy299) [reply build.gradle:2]: Ok, thanks

**Pull request merged**